packages/selected-network-controller/tests/SelectedNetworkController.test.ts

```diff
@@ -656,51 +709,63 @@ describe('SelectedNetworkController', () => {
     });
   });
 
-  describe('When a permission is added or removed', () => {
-    it('should add new domain to domains list on permission add if #useRequestQueuePreference is true', async () => {
-      const { controller, messenger } = setup({
-        useRequestQueuePreference: true,
-      });
-      const mockPermission = {
-        parentCapability: 'eth_accounts',
-        id: 'example.com',
-        date: Date.now(),
-        caveats: [{ type: 'restrictToAccounts', value: ['0x...'] }],
-      };
-
-      messenger.publish('PermissionController:stateChange', { subjects: {} }, [
-        {
-          op: 'add',
-          path: ['subjects', 'example.com', 'permissions'],
-          value: mockPermission,
-        },
-      ]);
+  describe('PermissionController:stateChange', () => {
+    describe('on permission add', () => {
+      describe('when useRequestQueuePreference is true', () => {
```

Comment:
Same point as above about nesting multiple `describe` blocks when the innermost one only wraps one test. I introduced similar patterns. I'm not personally too troubled about it but wonder what others feels about this @shanejonas @mcmire @Gudahtt ?

Reply:
I agree with being cautious about nesting `describe` blocks too much. After 3 or 4 is where I start to get confused as well (looking back, I made this mistake while writing NetworkController tests and I would love to clean those up at some point).

Reply:
specs have been flattened where possible

https://github.com/MetaMask/core/pull/4409/commits/b51a7ef7f98756eaa265d19c337cfd6b6307b6d8